lua/network_promises/http.lua

```diff
@@ -0,0 +1,108 @@
+NP.http = {}
+
+-- Returns a promise that resolves after t seconds
+-- fail : Should the promise reject after timeout
+function NP.timeout( t, fail )
+    local d = promise.new()
+    local method = fail and d.reject or d.resolve
+    timer.Simple( t, function()
+        method( d, "Timeout" )
+    end )
+    return d
+end
+
+local function responseSuccess( d, body, status, headers )
+    if math.floor( status / 100 ) == 2 then
+        d:resolve( body, status, headers )
+    else
+        d:reject( body, status, headers )
+    end
+end
+
+-- http.post as a promise, resolves whenever http finishes, or never if it doesn't ( Looking at you, ISteamHTTP )
+-- url    : post url
+-- data   : post args as table
+-- resolves to function( data, statusCode, headers )
+function NP.http.postIndef( url, data )
+    local d = promise.new() -- promise itself
+    http.Post( url, data, function( body, len, headers, status )
+        responseSuccess( d, body, status, headers )
+    end, function( err )
+        d:reject( err, -1 )
+    end )
+    return d
+end
+
+-- Same as above but for fetch
+function NP.http.fetchIndef( url )
+    local d = promise.new()
+    http.Fetch( url, function( body, len, headers, status )
+        responseSuccess( d, body, status, headers )
+    end, function( err )
+        d:reject( err, -1 )
+    end )
+    return d
+end
+
+-- HTTP as a promise, resolves whenever http finishes, or never if it doesn't (Looking at you, ISteamHTTP)
+-- method    : GET, PUT, etc.
+-- url       : 
+-- overrides : Table to be merged with the http struct (includes authToken as Token alias)
+-- resolves to function( data, statusCode, headers )
+function NP.http.requestIndef( method, url, overrides )
+    method = method or "GET"
+    local d = promise.new()
+    overrides = overrides or {}
+
+    local struct = {
+        failed = function( err )
+            d:reject( err, -1 )
+        end,
+        success = function( status, body, headers )
```

Comment:
Some of the other functions took `len` as the second or third param on `success` - is that not the case here, too?

Reply:
Request uses HTTP, others use http.Post, etc. While I'm not absolutely positive, this is Garry's mod here, inconsistency is like it's thing

Reply:
https://wiki.facepunch.com/gmod/Structures/HTTPRequest
It is status, body, headers